administrator/components/com_associations/views/association/tmpl/edit.php

```diff
@@ -14,15 +14,14 @@
 JHtml::_('formbehavior.chosen', 'select');
 
 $this->app->getDocument()->addScriptDeclaration("
-	function iframeRef( frameRef ) {
-	return frameRef.contentWindow
-		? frameRef.contentWindow.document
-		: frameRef.contentDocument
-	}
-
-	function triggerSave() {
-		var inside = iframeRef( document.getElementById('target-association') );
-		inside.getElementById('applyBtn').click();
+	function triggerSave(frame) {
+		if (frame == 'reference') {
+			window.frames[0].Joomla.submitbutton('" . $this->associatedView . ".apply');
+		}
+		if (frame == 'target') {
+			window.frames[1].Joomla.submitbutton('" . $this->associatedView . ".apply');
+			window.frames[0].location.reload();
```

Comment:
ok... so what if i add changed any of the reference language fields? will i lost all my changes, right?
So you have to options. 
1. Intead of reloading, add the reference dinamycly to the reference iframe associations tab. 
2. Use reload, but use a confirm dialog or something here.

My vote clearly  goes to 1. because of UX.